dev.php

```diff
@@ -54,43 +60,77 @@ function shortcode_ui_dev_advanced_example() {
 	 */
 	add_shortcode( 'shortcake_dev', 'shortcode_ui_dev_shortcode' );
 
+	if ( ! function_exists( 'shortcode_ui_register_for_shortcode' ) ) {
+		return;
+	}
+
 	/**
-	 * Register a UI for the Shortcode.
-	 * Pass the shortcode tag (string)
-	 * and an array or args.
+	 * Register UI for your shortcode
+	 *
+	 * @param string $shortcode_tag
+	 * @param array $ui_args
 	 */
-	shortcode_ui_register_for_shortcode(
-		'shortcake_dev', array(
-			'label' => 'Shortcake Dev', // Display label. String. Required.
-			'listItemImage' => 'dashicons-editor-quote', // Icon/attachment for shortcode. Optional. src or dashicons-$icon. Defaults to carrot.
+	shortcode_ui_register_for_shortcode( 'shortcake_dev',
+		array(
+			/*
+			 * How the shortcode should be labeled in the UI. Required argument.
+			 */
+			'label' => esc_html__( 'Shortcake Dev', 'shortcode-ui' ),
+			/*
+			 * Include an icon with your shortcode. Optional.
+			 * Use a dashicon, or full URL to image.
+			 */
+			'listItemImage' => 'dashicons-editor-quote',
+			/*
+			 * Limit this shortcode UI to specific posts. Optional.
+			 */
+			'post_type' => array( 'post' ),
+			/*
+			 * Register UI for the "inner content" of the shortcode. Optional.
+			 * If no UI is registered for the inner content, then any inner content
+			 * data present will be backed up during editing.
+			 */
 			'inner_content' => array(
-				'label' => 'Quote',
+				'label'        => esc_html__( 'Quote', 'shortcode-ui' ),
+				'description'  => esc_html__( 'Include a statement from someone famous.', 'shortcode-ui' ),
 			),
-			'post_type' => array( 'post' ), //Post type support
-			// Available shortcode attributes and default values. Required. Array.
-			// Attribute model expects 'attr', 'type' and 'label'
-			// Supported field types: text, checkbox, textarea, radio, select, email, url, number, and date.
+			/*
+			 * Register UI for attributes of the shortcode. Optional.
+			 * If no UI is registered for an attribute, then the attribute value will be backed up during editing.
```

Comment:
Hmm. This doesn't sound intuitive to me. I think it might read better like "If no UI is registered for an attribute, then the attribute will not be editable through Shortcake's UI. However, the value of any unregistered attributes will be preserved when editing."